### PR TITLE
[Frontend][MXNet] Change mxnet graph traversal from recursion to iteration

### PR DESCRIPTION
**Pull request by icemelon9**
Previous _from_mxnet_impl uses recursion to traverse graph. It causes call stack overflow when there're too many symbols in the graph. Now create symbols using topological order.

### CODE REVIEW

File: nnvm/python/nnvm/frontend/mxnet.py [hunk 1]
```diff
@@ -398,28 +447,29 @@ def _from_mxnet_impl(symbol, graph):
...
+        if childs is not None:
+            childs = [get_node(child) for child in childs]
+            childs = [x for y in childs for x in _as_list(y)]
+            node = _convert_symbol(op_name, childs, attr)
+        elif op_name != 'null':
+            node = _convert_symbol(op_name, [], attr)
+        else:
+            node = _sym.Variable(name=name, **attr)
+        graph[name] = node
+    return [get_node(sym) for sym in symbol]
```
Comment (zhreshold) [inline nnvm/python/nnvm/frontend/mxnet.py]: list is not a valid symbol, it's actually a GroupSymbol?  Also here should assert if node is not None

Comment (icemelon9) [reply nnvm/python/nnvm/frontend/mxnet.py]: Fixed

**Comment by tqchen** [comment]
Thanks @icemelon9 , @zhreshold can you take another look?

**Comment by zhreshold** [comment]
@icemelon9 Thanks, LGTM now.

**Comment by tqchen** [comment]
Thanks @icemelon9 @zhreshold ! this is now merged

**Pull request merged**